### ISSUE TITLE
Open the app on a specific screen Android

### ISSUE THREAD

**Issue by manuTro** [zo0r/react-native-push-notification#642]
Hi, I need the app to be open on a specific screen when the user clicks on the notification (I don't want button on my notifications), when the app is on background. 
I was able to make it works on iOS, but it's not working on Android. 
How I can do?

Thx

**Comment by fahadhaq** [comment]
Hi,

could you please share how you managed to open the app to specific screen for iOS. I am currently trying to do that for iOS but haven't got it to work yet. Any help would be greatly appreciated

**Comment by leofontes** [comment]
@manuTro @fahadhaq any of you ever work this out??

**Comment by fahadhaq** [comment]
Apologies for not posting the solution here. This is what I ended up doing.


Created a new function with callback

```
export function setupPushNotification(handleNotification) {
    PushNotification.configure({
        // (required) Called when a remote or local notification is opened or received
        onNotification: function (notification) {
            console.log('NOTIFICATION:', notification);

            if (notification.foreground) {
                if(notification.userInteraction)
                {
                    console.log('NOTIFICATION touched:', notification);
                    handleNotification(notification)
                }
                else{
                    console.log('NOTIFICATION foreground userInteraction:', notification.userInteraction);
                    handleNotification(notification)
                }
            }
            else {
                if(notification.userInteraction)
                {
                    console.log('NOTIFICATION touched:', notification);
                    handleNotification(notification)
                }
                else{
                    console.log('NOTIFICATION userInteraction:', notification.userInteraction);
                    handleNotification(notification)
                }
                
            }

            // required on iOS only (see fetchCompletionHandler docs: https://facebook.github.io/react-native/docs/pushnotificationios.html)
            notification.finish(PushNotificationIOS.FetchResult.NoData);
        },

        // ANDROID ONLY: GCM Sender ID (optional - not required for local notifications, but is need to receive remote push notifications)
        senderID: "",

        // IOS ONLY (optional): default: all - Permissions to register.
        permissions: {
            alert: true,
            badge: true,
            sound: true
        },

        // Should the initial notification be popped automatically
        // default: true
        popInitialNotification: true,
    });

    return PushNotification
}
```

then I added the following in my view 

```
componentDidMount() {
    this.pushNotification = setupPushNotification(this._handleNotificationOpen)
  }
```

Also the following in the same file

```
_handleNotificationOpen = (notification) => {
    try {
      if (Platform.OS === 'ios') {
        if (notification.data.news_or_deal.indexOf('d') !== -1) {
          const { navigate } = this.props.navigation
          navigate('DealsDetailsScreen', {
            type: 'Navigate',
            routeName: 'DealsDetailsScreen',
            params: {
              itemid: notification.data.news_or_deal_id
            }
          })
        }
        else if (notification.data.news_or_deal.indexOf('n') !== -1) {
          const { navigate } = this.props.navigation
          navigate('NewsDetailsScreen', {
            type: 'Navigate',
            routeName: 'NewsDetailsScreen',
            params: {
              itemid: notification.data.news_or_deal_id,
              itemtype: 'comp'
            }
          })
        }
      }
} catch (error) {
      Alert.alert('error', error)
    }
  }
```

I had to pass in extra values in the push notification.

For android I had to create a local notification regardless if the app was in foreground or background.

Let me know if this helps.

**Comment by ravindra4me** [comment]
Hello,

I am also facing the same issue, how do i open the android screen from the notification itself.

Thanks

**Comment by schumannd** [comment]
I think this depends on what navigation/routing package you are using.

Look at e.g. [this solution](https://github.com/react-navigation/react-navigation/issues/742#issuecomment-405592288) for react-navigation

**Comment by Sanan4li** [comment]
I am using "react-navigation" as you know when you click on an action button it opens the home screen of your app.
What I did is "navigate to a screen by using this.props.navigation.navigate("Screen") " when the action buttons is clicked.

**Comment by schumannd** [comment]
@Sanan4li 
I am sorry, I don't understand your explanation. Can you share some code?

**Comment by Sanan4li** [comment]
> @Sanan4li
> I am sorry, I don't understand your explanation. Can you share some code?

if you are working with react native you probably using react-navigation to switch between screens...
as you know when you click on the action button when a notification is popped up it will take you to the home screen in the app...
so what I did is I navigate to a specific screen where I want to...
I added this in home screen...
```

    componentDidMount = ()=>{
        PushNotification.configure({
            // (required) Called when a remote or local notification is opened or received
            onNotification: notification => {
              if (notification.action === "View") {
              
                console.log("View Clicked");
               
                PushNotification.cancelAllLocalNotifications();
                this.props.navigation.navigate("Detail", {body:notification.quoteInfo.body, by:notification.quoteInfo.by});
             
              } 
            },
            popInitialNotification: true,
            requestPermissions: true
          });
    }
    
```
